### PR TITLE
The lance now has a very clear confirmation message on it.

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Cargo crates can now have a tgui warning to it.
The lance shuttle gets it.

## Why It's Good For The Game

Let's have this set in stone.

## Images of changes

```This shuttle destroys EVERYTHING in its path when docking. You should only order this shuttle if: 1. The station has a clear need for it, 2. You are a hijacker, or 3. You have ahelped and received permission. Failure to meet this will lead to punishment.```

## Testing

ordered normal crate.
ordered lance, got mesage

## Changelog
:cl:
tweak: The lance now has a very clear confirmation message on it.
/:cl:

### CODE REVIEW

File: code/modules/supply/supply_pack.dm [hunk 1]
```diff
@@ -43,6 +43,8 @@
 	var/list/announce_beacons = list()
 	/// List of names for being done in TGUI
 	var/list/ui_manifest = list()
+	/// If this variable is filled, this grants the user a special TGUI confirmation prompt about ordering this shuttle, before they can place the order. Similar to bag of holding.
+	var/are_you_sure_you_want_to_be_banned
```
Comment (DGamerL) [inline code/modules/supply/supply_pack.dm:47]: Maybe move this to `/datum/supply_packs/abstract/shuttle`?

Comment (DGamerL) [reply code/modules/supply/supply_pack.dm:47]: Nevermind I'm an idiot

**Review by DGamerL — APPROVED** [review]
LGTM

File: code/modules/supply/supply_console.dm [hunk 2]
```diff
@@ -275,13 +275,15 @@
 			if(!P.can_order())
 				to_chat(user, "<span class='warning'>That cannot be ordered right now. Please try again later.</span>")
 				return
-
+			if(P.are_you_sure_you_want_to_be_banned)
+				var/we_warned_you = tgui_alert(user, "[P.are_you_sure_you_want_to_be_banned]", "ARE YOU SURE?", list("No", "Yes"))
```
Comment (Burzah) [inline code/modules/supply/supply_console.dm:279]:
```suggestion
				var/we_warned_you = tgui_alert(user, "[P.are_you_sure_you_want_to_be_banned]", "ARE YOU SURE?", list("Yes", "No"))
```
Working on trying to standardized button formatting.

Comment (Qwertytoforty) [reply code/modules/supply/supply_console.dm:279]: fair, was copying the BOH one so people don't instantly click yes

Comment (Burzah) [reply code/modules/supply/supply_console.dm:279]: The plan is to get them all switched eventually, as we now have a preference option that toggles the button position.

Comment (Qwertytoforty) [reply code/modules/supply/supply_console.dm:279]: Fair enough!

**Review by Burzah — APPROVED** [review]

**Pull request merged**